### PR TITLE
Introduce Traffic Generator container

### PR DESCRIPTION
**Pull request by RamLavi**
Currently the checkup is using a preset container image that runs the trex application on the pod.

Introducing the container image, as well as the scripts that configure the trex application and run it.
The parameters for this application are set by the checkup's params during the pod's run.
The pushed trex image is set as a build arg during the checkup's docker build process so that it can use the updated image.

### CODE REVIEW

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1675070570

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.002 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [75.588 seconds]
------------------------------

Ran 1 of 1 Specs in 75.591 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (75.59s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	75.623s
```

File: trex-config/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM quay.io/centos/centos:stream9
+
+ARG TREX_VERSION=2.87
+ENV TREX_VERSION ${TREX_VERSION}
+
+# install requirements
+RUN dnf -y install --nodocs git wget procps python3 vim python3-pip pciutils gettext https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && dnf clean all
+RUN dnf install -y --nodocs hostname iproute net-tools ethtool nmap iputils perf numactl sysstat htop rdma-core-devel libibverbs libibverbs-devel net-tools && dnf clean all
```
Comment (orelmisan) [inline trex-config/Dockerfile:8]: Please consider using a single RUN stanza, with escaped new line characters.
Using two RUN stanzas, creates two layers in the image (and needs to be cleaned twice).

File: trex-config/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+# install trex server
+WORKDIR /opt/
+RUN wget --no-check-certificate https://trex-tgn.cisco.com/trex/release/v${TREX_VERSION}.tar.gz && \
+    tar -xzf v${TREX_VERSION}.tar.gz && \
+    mv v${TREX_VERSION} trex && \
+    rm v${TREX_VERSION}.tar.gz
```
Comment (orelmisan) [inline trex-config/Dockerfile:15]: I don't see a build here, why is `git` required?
Why development libraries are required? (rdma-core-devel, libibverbs-devel)

File: trex-config/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+WORKDIR /opt/trex
```
Comment (orelmisan) [inline trex-config/Dockerfile:20]: Should this container run as root? or as a regular user?

File: Makefile [hunk 4]
```diff
@@ -31,12 +34,22 @@ build:
...
 
-push:
+push: push-trex
 	$(CRI_BIN) push $(REG)/$(ORG)/$(CHECKUP_IMAGE_NAME):$(CHECKUP_IMAGE_TAG)
 	$(CRI_BIN) tag $(REG)/$(ORG)/$(CHECKUP_IMAGE_NAME):$(CHECKUP_IMAGE_TAG) $(REG)/$(ORG)/$(CHECKUP_IMAGE_NAME):$(CHECKUP_GIT_TAG)
 	$(CRI_BIN) push $(REG)/$(ORG)/$(CHECKUP_IMAGE_NAME):$(CHECKUP_GIT_TAG)
 .PHONY: push
 
+build-trex:
+	$(CRI_BIN) build -f trex-config/Dockerfile -t $(REG)/$(ORG)/$(TREX_IMAGE_NAME):$(TREX_IMAGE_TAG) trex-config
```
Comment (orelmisan) [inline Makefile:44]: What is the meaning of `trex-config` at the end of this line?

File: pkg/internal/checkup/checkup.go [hunk 5]
```diff
@@ -261,14 +261,16 @@ func newTrafficGeneratorPod(checkupConfig config.Config, secondaryNetworkRequest
 	)
 
 	envVars := map[string]string{
-		config.PortBandwidthGBParamName: fmt.Sprintf("%d", checkupConfig.PortBandwidthGB),
+		"PORT_BANDWIDTH_GB":    fmt.Sprintf("%d", checkupConfig.PortBandwidthGB),
+		"NUMA_SOCKET":          fmt.Sprintf("%d", checkupConfig.NUMASocket),
+		"SET_VERBOSE":          "FALSE",
```
Comment (orelmisan) [inline pkg/internal/checkup/checkup.go:275]: Please consider using constants as env vars names.

**Review by orelmisan — COMMENTED** [review]
Thank you for the PR @RamLavi.
Reviewed the first three commits.

File: trex-config/Dockerfile (hunk 2 above)
Comment (RamLavi) [reply trex-config/Dockerfile:15]: >I don't see a build here, why is git required?

I'm not sure actually, will try to remove it

> Why development libraries are required? (rdma-core-devel, libibverbs-devel)

these are needed for the trex application.

File: trex-config/Dockerfile (hunk 3 above)
Comment (RamLavi) [reply trex-config/Dockerfile:20]: it is running as user 0 by the pod itself

Comment (RamLavi) [reply trex-config/Dockerfile:20]: I will mention in the commit message that this container will run with user 0

File: Makefile (hunk 4 above)
Comment (RamLavi) [reply Makefile:44]: that's the folder where the Dockerfile and other scripts reside

File: trex-config/Dockerfile (hunk 1 above)
Comment (RamLavi) [reply trex-config/Dockerfile:8]: DONE

File: trex-config/Dockerfile (hunk 2 above)
Comment (RamLavi) [reply trex-config/Dockerfile:15]: removed git, the devel ones, htop and the rpm one (was required for hop)

File: pkg/internal/checkup/checkup.go (hunk 5 above)
Comment (RamLavi) [reply pkg/internal/checkup/checkup.go:275]: DONE

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1675108807

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [75.602 seconds]
------------------------------

Ran 1 of 1 Specs in 75.606 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (75.61s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	75.638s
```

trex daemon seems to be running
```
link : link : Link Up - speed 10000 Mbps - full-duplex
promiscuous : 0
port : 1
------------
link : link : Link Up - speed 10000 Mbps - full-duplex
promiscuous : 0
number of ports : 2
max cores for 2 ports : 6
tx queues per port : 8
-------------------------------
RX core uses TX queue number 65535 on all ports
core, c-port, c-queue, s-port, s-queue, lat-queue

```

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1675108807

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [75.602 seconds]
------------------------------

Ran 1 of 1 Specs in 75.606 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (75.61s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	75.638s
```

trex daemon seems to be running
```
...
link : link : Link Up - speed 10000 Mbps - full-duplex
promiscuous : 0
port : 1
------------
link : link : Link Up - speed 10000 Mbps - full-duplex
promiscuous : 0
number of ports : 2
max cores for 2 ports : 6
tx queues per port : 8
-------------------------------
RX core uses TX queue number 65535 on all ports
core, c-port, c-queue, s-port, s-queue, lat-queue

```

**Comment by RamLavi** [comment]
Change: [Rebase](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/a66e85ee860b87955f4e741416b87d622a313a1c..dd1a931b1dac4563091d9be5f41dd0380a60a0f0)

File: trex-config/templates/trex_cfg.yaml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+- port_limit: 2
+  version: 2
```
Comment (RamLavi) [inline trex-config/templates/trex_cfg.yaml:2]: change name suffix

File: pkg/internal/checkup/checkup.go [hunk 7]
```diff
@@ -256,19 +256,24 @@ func (c *Checkup) waitForPodRunningStatus(ctx context.Context, namespace, name s
 
 func newTrafficGeneratorPod(checkupConfig config.Config, secondaryNetworkRequest string) *k8scorev1.Pod {
 	const (
-		TrexPodNumCPUs      = "8"
-		TrexPodNumHugepages = "8Gi"
+		TrexPodNumCPUs         = "8"
```
Comment (RamLavi) [inline pkg/internal/checkup/checkup.go]: add new line

File: trex-config/scripts/main.sh [hunk 8]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (RamLavi) [inline trex-config/scripts/main.sh:1]: add licneces

File: trex-config/templates/trex_cfg.yaml [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+  interfaces:
+    - $PCIDEVICE_OPENSHIFT_IO_DPDK_NIC_1
```
Comment (RamLavi) [inline trex-config/templates/trex_cfg.yaml:4]: remove openshift name

File: trex-config/scripts/set_trex_cfg_file.sh [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+  echo MASTER="${MASTER}"
+  echo LATENCY="${LATENCY}"
+  echo NUMA_SOCKET="${NUMA_SOCKET}"
+  echo CPU="${CPU}"
+  echo PORT_BANDWIDTH_GB="${PORT_BANDWIDTH_GB}"
+}
+
+set_pci_addresses
+
+set_numa
```
Comment (RamLavi) [inline trex-config/scripts/set_trex_cfg_file.sh:95]: remove cause set on checkup

File: trex-config/scripts/set_trex_cfg_file.sh [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+  # set cpu
+  CPU=${CPUS}
+  CPU=$(remove_number "${CPU}" "${MASTER}")
+  CPU=$(remove_number "${CPU}" "${LATENCY}")
+  export CPU=$CPU
+}
+
+print_params() {
+	echo setting params:
```
Comment (RamLavi) [inline trex-config/scripts/set_trex_cfg_file.sh:83]: fix indent

File: trex-config/scripts/set_trex_cfg_file.sh [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+set_cpu_configs
+
+print_params
+
+# set config_template file
+envsubst < /opt/templates/trex_cfg.yaml > /etc/trex_cfg.yaml
+
+if [ "${SET_VERBOSE}" == "TRUE" ]; then
+	cat /etc/trex_cfg.yaml
```
Comment (RamLavi) [inline trex-config/scripts/set_trex_cfg_file.sh:105]: always print

File: trex-config/scripts/set_trex_cfg_file.sh [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+remove_number() {
+	local list=$1
+	local number=$2
+	local new_list=""
+	IFS=',' read -r -a array <<< "$list"
+	for i in "${array[@]}"; do
+	if [ "$i" != "$number" ]; then
+	new_list="$new_list,$i"
```
Comment (RamLavi) [inline trex-config/scripts/set_trex_cfg_file.sh:32]: fix indent

File: trex-config/scripts/set_trex_cfg_file.sh [hunk 14]
```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
```
Comment (RamLavi) [inline trex-config/scripts/set_trex_cfg_file.sh:1]: change stanza

File: trex-config/scripts/set_trex_cfg_file.sh [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+	if [ "$i" != "$number" ]; then
+	new_list="$new_list,$i"
+	fi
+	done
+	new_list=${new_list:1} # remove the leading comma
+	echo "${new_list}"
+}
+
+get_pci_device_env_var() {
+	local pci_device_env_with_value=$(env | grep "PCIDEVICE_OPENSHIFT_IO")
```
Comment (RamLavi) [inline trex-config/scripts/set_trex_cfg_file.sh:40]: change to PCIDEVICE_

File: pkg/internal/checkup/checkup.go [hunk 16]
```diff
@@ -256,17 +256,30 @@ func (c *Checkup) waitForPodRunningStatus(ctx context.Context, namespace, name s
...
-		portBandwidthParamName = "PORT_BANDWIDTH_GB"
-		numaSocketParamName    = "NUMA_SOCKET"
-		verboseParamName       = "SET_VERBOSE"
+		trexPodNumCPUs             = 8
+		trexNumOfNonTrafficCpus    = 2
+		trexPodNumHugepages        = "8Gi"
+		portBandwidthParamName     = "PORT_BANDWIDTH_GB"
+		numaSocketParamName        = "NUMA_SOCKET"
+		verboseParamName           = "SET_VERBOSE"
+		numTrafficCpusParamName    = "NUM_OF_TRAFFIC_CPUS"
```
Comment (RamLavi) [inline pkg/internal/checkup/checkup.go:265]: move to next commit

**Comment by RamLavi** [comment]
```
zeus09:kubevirt-dpdk-checkup $ TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1675177125

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:56
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [70.682 seconds]
------------------------------

Ran 1 of 1 Specs in 70.686 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (70.69s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	70.717s
```

**Comment by RamLavi** [comment]
Change: [Rebase](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/d057c15ec8562c487d779b79b05806a5edfde2bc..15e35bd3fef46b5e4788efa181f395d850ea5582) with conflict fixes

**Comment by RamLavi** [comment]
Change: [Rebase](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/15e35bd3fef46b5e4788efa181f395d850ea5582..f9290694ff59a6d18433ac5ba5a3902482a8d24f)

**Comment by RamLavi** [comment]
manually pushed the created traffic-generator container and ran it with it: `quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:initial`
```
zeus09:kubevirt-dpdk-checkup $ TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:initial TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           -e RUNTIME_CLASS_NAME= \
           -e TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:initial \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1675255746

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:61
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [95.747 seconds]
------------------------------

Ran 1 of 1 Specs in 95.751 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (95.75s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	95.782s
```

**Comment by RamLavi** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/f7e7d8517133f5da594fb2a57c525a882dfd3339..3d4c446cff1ed2c0d2f7ba17c319e817e3ad29a0): Rebase


```
zeus09:kubevirt-dpdk-checkup $ TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:initial TEST_NAMESPACE=ralavi-ns TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest make test/e2e
docker run --rm \
           --volume /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup:Z \
           --volume /root/.kube:/root/.kube:Z \
           --workdir /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup \
           -e KUBECONFIG=/root/.kube/config \
           -e TEST_IMAGE=quay.io/ramlavi/kubevirt-dpdk-checkup:latest \
           -e TEST_NAMESPACE=ralavi-ns \
           -e NETWORK_ATTACHMENT_DEFINITION_NAME= \
           -e RUNTIME_CLASS_NAME= \
           -e TRAFFIC_GEN_IMAGE_URL=quay.io/ramlavi/kubevirt-dpdk-checkup-traffic-gen:initial \
           docker.io/library/golang:1.19 go test ./tests/... -test.v -test.timeout=1h -ginkgo.v -ginkgo.timeout=1h
=== RUN   TestKubevirtDpdkCheckup
Running Suite: KubevirtDpdkCheckup Suite - /root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests
=======================================================================================================
Random Seed: 1675257252

Will run 1 of 1 specs
------------------------------
[BeforeSuite] 
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/test_suite_test.go:61
[BeforeSuite] PASSED [0.004 seconds]
------------------------------
Execute the checkup Job should complete successfully
/root/go/src/github.com/kiagnose/kubevirt-dpdk-checkup/tests/checkup_test.go:79
• [105.727 seconds]
------------------------------

Ran 1 of 1 Specs in 105.732 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
--- PASS: TestKubevirtDpdkCheckup (105.73s)
PASS
ok  	github.com/kiagnose/kubevirt-dpdk-checkup/tests	105.762s
```

File: traffic-gen-config/Dockerfile [hunk 17]
```diff
@@ -0,0 +1,34 @@
+FROM quay.io/centos/centos:stream9
```
Comment (orelmisan) [inline traffic-gen-config/Dockerfile:1]: Could you please rename the directory to `traffic-gen`?

File: traffic-gen-config/scripts/main.sh [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2023 Red Hat, Inc.
+#
+
```
Comment (orelmisan) [inline traffic-gen-config/scripts/main.sh:19]: Please consider adding
`set -e`
`set -u`
to stop on error and when variables are used but were not defined.

File: traffic-gen-config/scripts/main.sh [hunk 19]
```diff
@@ -20,3 +20,6 @@
 echo "setting params to trex_cfg.yaml"
 /opt/scripts/set_traffic_gen_cfg_file.sh
 
+echo "setting params to traffic-gen test files"
+/opt/scripts/set_tests_files.sh
+
```
Comment (orelmisan) [inline traffic-gen-config/scripts/main.sh:25]: nit: redundant new line.

File: traffic-gen-config/templates/testpmd.py.in [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+mac_localport1="$SRC_EAST_MAC_ADDRESS"
+
+class STLS1(object):
+
+    def __init__ (self):
+        self.fsize  =64; # the size of the packet
+        self.number = 0
+
+    def create_stream (self, direction = 0):
+
```
Comment (orelmisan) [inline traffic-gen-config/templates/testpmd.py.in:16]: nit: redundant new line.

File: pkg/internal/checkup/checkup.go [hunk 21]
```diff
@@ -289,12 +289,14 @@ func (c *Checkup) waitForPodRunningStatus(ctx context.Context, namespace, name s
 
 func newTrafficGeneratorPod(checkupConfig config.Config, secondaryNetworkRequest string) *k8scorev1.Pod {
 	const (
-		trafficGeneratorPodCPUCount       = 8
-		trafficGeneratorPodHugepagesCount = "8Gi"
+		trafficGeneratorPodCPUCount            = 8
+		trafficGeneratorPodNumOfNonTrafficCpus = 2
```
Comment (orelmisan) [inline pkg/internal/checkup/checkup.go:293]: nit:
```suggestion
		trafficGeneratorPodNumOfNonTrafficCPUs = 2
```

File: traffic-gen-config/scripts/run_traffic_gen_daemon.sh [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+	set -x
+fi
+
+print_params() {
+echo NUM_OF_TRAFFIC_CPUS="${NUM_OF_TRAFFIC_CPUS}"
+}
+
+print_params
+
+./t-rex-64 --no-ofed-check --no-hw-flow-stat -i -c "${NUM_OF_TRAFFIC_CPUS}" --iom 0
```
Comment (orelmisan) [inline traffic-gen-config/scripts/run_traffic_gen_daemon.sh:30]: Could you please add some details about this command to the commit message?

File: traffic-gen-config/scripts/run_traffic_gen_daemon.sh [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+#
+# Copyright 2023 Red Hat, Inc.
+#
+
+if [ "${SET_VERBOSE}" == "TRUE" ]; then
+	set -x
+fi
+
+print_params() {
+echo NUM_OF_TRAFFIC_CPUS="${NUM_OF_TRAFFIC_CPUS}"
```
Comment (orelmisan) [inline traffic-gen-config/scripts/run_traffic_gen_daemon.sh:25]: nit: missing indentation.

File: traffic-gen-config/scripts/main.sh (hunk 19 above)
Comment (RamLavi) [reply traffic-gen-config/scripts/main.sh:25]: DONE

File: traffic-gen-config/templates/testpmd.py.in (hunk 20 above)
Comment (RamLavi) [reply traffic-gen-config/templates/testpmd.py.in:16]: DONE

File: pkg/internal/checkup/checkup.go (hunk 21 above)
Comment (RamLavi) [reply pkg/internal/checkup/checkup.go:293]: DONE

File: traffic-gen-config/scripts/run_traffic_gen_daemon.sh (hunk 23 above)
Comment (RamLavi) [reply traffic-gen-config/scripts/run_traffic_gen_daemon.sh:25]: DONE

File: traffic-gen-config/scripts/run_traffic_gen_daemon.sh (hunk 22 above)
Comment (RamLavi) [reply traffic-gen-config/scripts/run_traffic_gen_daemon.sh:30]: DONE

File: traffic-gen-config/scripts/main.sh (hunk 18 above)
Comment (RamLavi) [reply traffic-gen-config/scripts/main.sh]: DONE for all scripts

**Comment by RamLavi** [comment]
[Change](https://github.com/kiagnose/kubevirt-dpdk-checkup/compare/3d4c446cff1ed2c0d2f7ba17c319e817e3ad29a0..a6ae43f2b06af8aa3a70d578c45b6b8ee7903de2): review fixes

**Review by orelmisan — APPROVED** [review]
Thank you for the changes

**Comment by orelmisan** [comment]
As we have talked offline, IMO we should replace these scripts with a higher level language when the opportunity arises.

**Pull request merged**